lib/client.js

```diff
@@ -382,872 +462,865 @@ const warmupLimit = 32;
-     * @type {ClientMetrics}
+     * Emitted when a host in the cluster changed status from down to up.
+     * <ul>
+     *   <li>{@link Host host} The host that changed the status.</li>
+     * </ul>
+     * @event Client#hostUp
+     */
+    /**
+     * Emitted when a host in the cluster changed status from up to down.
+     * <ul>
+     *   <li>{@link Host host} The host that changed the status.</li>
+     * </ul>
+     * @event Client#hostDown
      */
-    this.metrics = this.options.metrics;
-}
-
-util.inherits(Client, events.EventEmitter);
 
-/**
- * Emitted when a new host is added to the cluster.
- * <ul>
- *   <li>{@link Host} The host being added.</li>
- * </ul>
- * @event Client#hostAdd
- */
-/**
- * Emitted when a host is removed from the cluster
- * <ul>
- *   <li>{@link Host} The host being removed.</li>
- * </ul>
- * @event Client#hostRemove
- */
-/**
- * Emitted when a host in the cluster changed status from down to up.
- * <ul>
- *   <li>{@link Host host} The host that changed the status.</li>
- * </ul>
- * @event Client#hostUp
- */
-/**
- * Emitted when a host in the cluster changed status from up to down.
- * <ul>
- *   <li>{@link Host host} The host that changed the status.</li>
- * </ul>
- * @event Client#hostDown
- */
+    /**
+     * Attempts to connect to one of the [contactPoints]{@link ClientOptions} and discovers the rest the nodes of the
+     * cluster.
+     * <p>When the {@link Client} is already connected, it resolves immediately.</p>
+     * <p>It returns a <code>Promise</code> when a <code>callback</code> is not provided.</p>
+     * @param {function} [callback] The optional callback that is invoked when the pool is connected or it failed to
+     * connect.
+     * @example <caption>Usage example</caption>
+     * await client.connect();
+     */
+    connect(callback) {
+        if (this.connected && callback) {
+            // Avoid creating Promise to immediately resolve them
+            return callback();
+        }
 
-/**
- * Attempts to connect to one of the [contactPoints]{@link ClientOptions} and discovers the rest the nodes of the
- * cluster.
- * <p>When the {@link Client} is already connected, it resolves immediately.</p>
- * <p>It returns a <code>Promise</code> when a <code>callback</code> is not provided.</p>
- * @param {function} [callback] The optional callback that is invoked when the pool is connected or it failed to
- * connect.
- * @example <caption>Usage example</caption>
- * await client.connect();
- */
-Client.prototype.connect = function (callback) {
-    if (this.connected && callback) {
-        // Avoid creating Promise to immediately resolve them
-        return callback();
+        return promiseUtils.optionalCallback(this._connect(), callback);
     }
 
-    return promiseUtils.optionalCallback(this._connect(), callback);
-};
+    /**
+     * Async-only version of {@link Client#connect()}.
+     * @private
+     */
+    async _connect() {
```

Comment:
If the _ character is used to denote a private method (I think this was a convention in older JS, but I am not sure) we can check if switching to a [newer syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties) would work.